lxd/storage/backend_lxd.go

```diff
@@ -1025,6 +1025,11 @@ func (b *lxdBackend) CreateInstanceFromCopy(inst instance.Instance, src instance
 	volStorageName := project.Instance(inst.Project().Name, inst.Name())
 	vol := b.GetVolume(volType, contentType, volStorageName, srcConfig.Volume.Config)
 
+	// Set a new UUID.
+	// Do this before asking the storage driver if it has the volume.
+	// Some storage drivers might require the UUID to generate the volume name.
+	vol.Config()["volatile.uuid"] = uuid.New().String()
```

Comment:
This seems unnecessary, if the volume doesn't have a uuid then it wont exist on storage drivers that need uuids. So the storage driver should just return false. Rather than requiring these special cases. Ideally storage driver specific cases should be handled in the storage driver.

Reply:
Maybe the comment is misleading here. 
`vol.Config()` already contains the UUID from the source volume (through `srcConfig.Volume.Config`). 
If we now call `b.driver.HasVolume(vol)`, the PowerFlex driver will create the source volume's name from the source volume's UUID instead of taking a fresh UUID whose volume doesn't yet exist on storage.

Reply:
That means if we don't set a fresh UUID here, `HasVolume` will tell that the volume already exists.

Reply:
This is now completely removed from the PR.